core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1910,7 +1931,7 @@ class Log(@volatile private var _dir: File,
         in the header.
       */
       appendInfo.firstOffset match {
-        case Some(firstOffset) => roll(Some(firstOffset))
+        case Some(firstOffset) => roll(Some(firstOffset.messageOffset))
```

Comment:
nit: might be able to simplify this a little bit. Maybe something like this:
```scala
val rollOffset = appendInfo.firstOffset.map(_.messageOffset)
  .getOrElse(maxOffsetInMessages - Integer.MAX_VALUE))
roll(Some(rollOffset))
```

Reply:
Done.